### ISSUE TITLE
Several pkg-config files with wrong linker flags

### ISSUE THREAD

**Issue by Febbe** [msys2/MINGW-packages#16362]
### Description / Steps to reproduce the issue

1. install gdk-3.0, gdk-broadway-3.0 gdk-win32-3.0
2. try to link a library/executable via cmake against gdk-3.0
3. A warning will be issued

### Expected behavior

There is no warning issued when linking against gdk

### Actual behavior

`[build] LINK : warning LNK4044: Nicht erkannte Option /Wl,-luuid; wird ignoriert.`
translated: `[build] LINK : warning LNK4044: Ignored unrecognized option /Wl,-luuid;.`

After running a python script, which scanned all .pc files I found the following:
`C:\msys64\ucrt64\lib\pkgconfig\gdk-3.0.pc`
`C:\msys64\ucrt64\lib\pkgconfig\gdk-broadway-3.0.pc`
`C:\msys64\ucrt64\lib\pkgconfig\gdk-win32-3.0.pc`

`gdk-3.0.pc`:
`Libs: -L${libdir} -lgdk-3 -lz -lgdi32 -limm32 -lshell32 -lole32 -Wl,-luuid -lwinmm -ldwmapi -lsetupapi -lcfgmgr32 -lhid -lwinspool -lcomctl32 -lcomdlg32`

Note the `,` in the line, which should be a space

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-22621

### MINGW environments affected

- [ ] MINGW64
- [ ] MINGW32
- [X] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

**Comment by MehdiChinoune** [comment]
Looking at the output, It seems that you are using MSVC.

**Comment by Biswa96** [comment]
Yeah, it looks like msvc from the log but the libraries may be used in other toolchain. @Febbe Could you make sure that changing `-Wl,-luuid` to `-luuid` fixes the issue?

**Comment by Febbe** [comment]
Yes, I am trying some hard stuff with msvc and mixed toolchains. But that's not the point, the `,` does not belong there, also for gcc/clang. Removing it and the `-WL` fixed my issue.

**Comment by Febbe** [comment]
@MehdiChinoune just for the explanation, I want to use msvc or clang-cl, because they don't have some issues, gcc has on Windows. It is also better, regarding standard c++ compliance. Therefore, I tried vcpkg, but it seems they also just fetch msys, and copy the libs, but it crashed on runtime.   

I thought I can do better. And actually, after 
1. creating a C:/msys64/msvc directory
2. fixing all pkg-config files with the new directory, starting with C:/msys64/msvc
3. Removed all -lm flags from .pc files + fixed this issue 
4. deleting all static libraries
5. renaming all lib(.*)\.dll\.a libraries to $1.lib 
6. Added some paths to PATH

It finally worked. It seems like linking to dynamic libraries, generated by gcc with link.exe works just fine.

**Comment by MehdiChinoune** [comment]
> I want to use msvc or clang-cl, because they don't have some issues, gcc has on Windows. It is also better, regarding standard c++ compliance.

- We have CLANG64 environment if you want to use llvm-toolchain (llvm+clang+lld+...)
- What do you mean by "standard c++ compliance"?, GCC is known to be the first compiler to comply to new C++ standards 
https://en.cppreference.com/w/cpp/compiler_support/17
https://en.cppreference.com/w/cpp/compiler_support/20

**Comment by Biswa96** [comment]
The gtk3 pkgconfig related issue has been fixed with this commit https://github.com/msys2/MINGW-packages/commit/3a61ad53cf8b9baf9f677254dfce5146042b3ad5. gtk3 package version 3.24.37.r22.gf1b6556-2 or above has the fixed pkgconfig files. Please update the packages with `pacman -Syyu` command.

Thank you for reporting the issue.

**Comment by Febbe** [comment]
> - We have CLANG64 environment if you want to use llvm-toolchain (llvm+clang+lld+...)

The problem is, that llvm is slow, regarding the implementation of the c++ standard, they are still not done wit c++20. Also we have llvm17 in development and llvm17 released. But msys ships clang 11/12 with `pacman -Syu clang`

Another thing is the STL, I love using libc++, but it is still not done. And Microsoft's STL is both done and also fits better with the os.

When I could just use clang(-cl) with the Microsoft STL from within MSYS, I would be satisfied.

> - What do you mean by "standard c++ compliance"?, GCC is known to be the first compiler to comply to new C++ standards 

Depends on the view, MSVC was the first which was done with both C++20 and the STL.
And I believe they will also be the first for C++23.

**Comment by TheShermanTanker** [comment]
> @MehdiChinoune just for the explanation, I want to use msvc or clang-cl, because they don't have some issues, gcc has on Windows. It is also better, regarding standard c++ compliance. Therefore, I tried vcpkg, but it seems they also just fetch msys, and copy the libs, but it crashed on runtime.
> 
> I thought I can do better. And actually, after
> 
> 1. creating a C:/msys64/msvc directory
> 2. fixing all pkg-config files with the new directory, starting with C:/msys64/msvc
> 3. Removed all -lm flags from .pc files + fixed this issue
> 4. deleting all static libraries
> 5. renaming all lib(.*).dll.a libraries to $1.lib
> 6. Added some paths to PATH
> 
> It finally worked. It seems like linking to dynamic libraries, generated by gcc with link.exe works just fine.

While Visual C++ does better with Windows specific features that gcc isn't able to replicate too well (such as SEH handlers), it is really terrible when it comes to Standard C++ conformance. It even has a permissive- switch of which the only functionality is to force conforming code when compiling (not even to disable additional compiler specific features, long for example is a typedef for int when compiling normally!!)

There's a reason why the native Java libraries for Windows is so poor in code quality when compared to the ones on MacOS and Linux, Hint: Visual C++ is currently the only compiler available for Windows in the JDK, while MacOS and Linux can choose between gcc and clang ;)

**Comment by Febbe** [comment]
> While Visual C++ does better with Windows specific features that gcc isn't able to replicate too well (such as SEH handlers), 

Yes, actually that is what I wan't for the implementation of the STL. When I use c++ locales I expect them to work just like or similar on a posix system. But a very long time, the implementation of stdlibc++(GCC) couldn't assure this (actually I don't know if it's still the case). Or some new STL features like std:: filesystem had enourmous problems. Of cause I want to use the Microsoft STL then.

> it is really terrible when it comes to Standard C++ conformance. It even has a permissive- switch of which the only functionality is to force conforming code when compiling (not even to disable additional compiler specific features, long for example is a typedef for int when compiling normally!!)

Currently I Can't check if it is still true for Vs studio 2022. But the C++(I can't speak for C) standard only defines a non strict size relation.
And that they are buildin types (which are not part of any header). Microsoft itself mentioned in their documentation, that those types do not appear in headers, but might still be defined by the compiler as alias, which is completely fine. 
I think the 80's c++ committee is to blame here, because they let the compiler manufacturers chose any size they want. Now those want to ensure backward compatibility and have to keep all their mistakes...

The permissive flag is required and implicit for nearly every c++20 feature. And GCC has some sort of pedantic flag which is similar. And the thing which bothers me is, that GCC and clang accept code which is non c++17 conforming, but which is C-17 conforming in a c++ source file.

> There's a reason why the native Java libraries for Windows is so poor in code quality when compared to the ones on MacOS and Linux, Hint: Visual C++ is currently the only compiler available for Windows in the JDK, while MacOS and Linux can choose between gcc and clang ;)

How does the code quality depend on the compiler and not on the developer? I mean, you could test your code against multiple compilers which may have better diagnostics and are more pedantic regarding questionable language constructs, but that does not necessarily lead to good code quality.
